tests/vmi_configuration_test.go

```diff
@@ -2508,15 +2508,27 @@ var _ = Describe("Configurations", func() {
 		vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
 		Expect(err).ToNot(HaveOccurred(), "Should successfully get VMI")
 
-		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+		vmiVirtLauncherPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
 
 		tests.WaitAgentConnected(virtClient, vmi)
 
-		// Ensure that VMI IP Stays equal to PodIP even after Guest-Agent kicks in
+		// Ensure that VMI 'ipAddress' and 'ipAddresses' stays equal to
+		// VMI virt-launcher pod 'PodIP' and 'PodIPs' even after Guest-Agent kicks in
 		Consistently(func() bool {
 			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred(), "Should successfully get VMI")
-			return vmi.Status.Interfaces[0].IP == vmiPod.Status.PodIP
+
+			isPodIPSynced := vmiVirtLauncherPod.Status.PodIP == vmi.Status.Interfaces[0].IP
+
+			isPodIPsSynced := true
+			for idx := 0; idx < len(vmiVirtLauncherPod.Status.PodIPs); idx++ {
+				if vmiVirtLauncherPod.Status.PodIPs[idx].IP != vmi.Status.Interfaces[0].IPs[idx] {
```

Comment:
you iterate according `vmiVirtLauncherPod.Status.PodIP` len
and accessing `vmi.Status.Interfaces[0].IPs`
in case len of `vmi.Status.Interfaces[0].IPs` < len of `vmiVirtLauncherPod.Status.PodIP` there would be a panic,
i would suggest to add len check verify please

Reply:
You right, I will add a check for length of both arrays.

I intentionally dont use `range` because I am not sure there is grantee that it will return ascending sequence of `0,1..n`

Reply:
on maps it wont reserve order, on array / slice it does
and here even if it doesnt you dont need order